src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/UnsafeAccessorAttribute.cs

```diff
@@ -0,0 +1,89 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// The kind of target an <see cref="UnsafeAccessorAttribute" /> is
+    /// providing access.
+    /// </summary>
+    public enum UnsafeAccessorKind
+    {
+        /// <summary>
+        /// Provide access to a Constructor.
+        /// </summary>
+        Constructor,
+
+        /// <summary>
+        /// Provide access to a Method.
+        /// </summary>
+        Method,
+
+        /// <summary>
+        /// Provide access to a static Method.
+        /// </summary>
+        StaticMethod,
+
+        /// <summary>
+        /// Provide access to a Field.
+        /// </summary>
+        Field,
+
+        /// <summary>
+        /// Provide access to a static Field.
+        /// </summary>
+        StaticField
+    };
+
+    /// <summary>
+    /// Provide access to an unaccessible API on a specific type.
+    /// </summary>
+    /// <remarks>
+    /// This attribute should be applied on a <code>extern static</code> method.
+    /// The implementation of the <code>extern static</code> method annotated with
+    /// this attribute will be provided by the runtime based on the information in
+    /// the attribute and the signature of the method that the attribute is applied to.
+    /// The runtime will try to find the matching method or field and forward the call
+    /// to it. If the matching method or field is not found, the body of the <code>extern</code>
+    /// method will throw <see cref="MissingFieldException" /> or <see cref="MissingMethodException" />.
+    ///
+    /// For <see cref="UnsafeAccessorKind.Method"/>/<see cref="UnsafeAccessorKind.StaticMethod"/> and
+    /// <see cref="UnsafeAccessorKind.Field"/>/<see cref="UnsafeAccessorKind.StaticField"/>, the type of
+    /// the first argument of the annotated <code>extern</code> method identifies the owning type.
+    /// The value of the first argument is treated as <code>this</code> pointer for instance fields and methods.
+    /// The first argument must be passed as <code>ref</code> for instance fields and methods on structs.
+    /// The value of the first argument is not used by the implementation for <code>static</code> fields and methods.
+    ///
+    /// The generic parameters of the <code>extern static</code> method are concatenation of the type and
+    /// method generic arguments of the target method. For example,
+    /// <code>extern static void Method1&lt;T1, T2&gt;(Class1&lt;T1&gt; @this)</code>
+    /// can be used to call <code>Class1&lt;T1&gt;.Method1&lt;T2&gt;()</code>. The generic constraints of the
+    /// <code>extern static</code> method must match generic constraints of the target type, field or method.
```

Comment:
Can this extern static method be on a generic type?  If yes, how does that impact this.  If no, we should state so.